sdk/sdkerrors/errors.go

```diff
@@ -97,7 +93,36 @@ func AsConnectError(err error) error {
 		return connect.NewError(connect.CodeUnimplemented, err)
 	case errors.Is(err, ErrFailedPrecondition):
 		return connect.NewError(connect.CodeFailedPrecondition, err)
+	case errors.Is(err, ErrProgram):
+		fallthrough
 	default:
 		return connect.NewError(connect.CodeUnknown, err)
 	}
 }
+
+func ErrorType(err error) string {
```

Comment:
I prefer that each error (e.g. `ErrNotFound`) will implement a method (say `Type() string`). Then you can do something like:

```go
type Typer interface {
	Type() string
}

func ErrorType(e error) string {
	if t, ok := e.(Typer); ok {
		return t.Type()
	}

	return "unknown error"
}
```

Reply:
cool, fixed. ptal